src/core/include/openvino/op/is_inf.hpp

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/op/op.hpp"
+
+namespace ov {
+namespace op {
+namespace v10 {
+/// \brief Boolean mask that maps infinite values to true.
+/// \ingroup ov_ops_cpp_api
+class OPENVINO_API IsInf : public Op {
+public:
+    OPENVINO_OP("IsInf", "opset10");
+
+    /// \brief A Structure which contains all IsInf attributes
+    struct Attributes {
+        // A flag which specifies whether to map negative invinities to true.
+        // If set to false, negative infinite will be mapped to false.
+        bool detect_negative = true;
+        // A flag which specifies whether to map positive invinities to true.
+        // If set to false, positive infinite will be mapped to false.
+        bool detect_positive = true;
+
+        Attributes() = default;
+        Attributes(bool detect_negative, bool detect_positive)
+            : detect_negative{detect_negative},
+              detect_positive{detect_positive} {}
+    };
+
+    IsInf() = default;
+    /// \brief Constructs a IsInf operation
+    ///
+    /// \param data   Input data tensor
+    /// \param attrs  IsInf attributes
+    IsInf(const Output<Node>& data, const Attributes& attributes);
```

Comment:
As all of the attributes are specified as optional, it's worth to consider additional `IsInf(const Output<Node>& data);` constructor.
```suggestion
    IsInf(const Output<Node>& data, const Attributes& attributes);
    IsInf(const Output<Node>& data);
```

Reply:
Thanks, done